metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_aspects.py

```diff
@@ -0,0 +1,570 @@
+            logger.warning(
+                f"Dataset {dataset.path}.{dataset.resource_name} will have a null schema"
+            )
+
+        # Status
+        status = StatusClass(removed=False)
+        mcp = MetadataChangeProposalWrapper(
+            entityUrn=dataset_urn,
+            aspect=status,
+        )
+        yield mcp.as_workunit()
+
+    def populate_glossary_term_mcp(
+        self, glossary_term: DremioGlossaryTerm
+    ) -> Iterable[MetadataWorkUnit]:
+        glossary_term_info = self._create_glossary_term_info(glossary_term)
+        mcp = MetadataChangeProposalWrapper(
+            entityUrn=glossary_term.urn,
+            aspect=glossary_term_info,
+        )
+        yield mcp.as_workunit()
+
+    def populate_profile_aspect(self, profile_data: Dict) -> DatasetProfileClass:
+        field_profiles = [
+            self._create_field_profile(field_name, field_stats)
+            for field_name, field_stats in profile_data.get("column_stats", {}).items()
+        ]
+        return DatasetProfileClass(
+            timestampMillis=round(time.time() * 1000),
+            rowCount=profile_data.get("row_count"),
+            columnCount=profile_data.get("column_count"),
+            fieldProfiles=field_profiles,
+        )
+
+    def _create_container_properties(
+        self, container: DremioContainer
+    ) -> ContainerPropertiesClass:
+        return ContainerPropertiesClass(
+            name=container.container_name,
+            qualifiedName=f"{'.'.join(container.path) + '.' if container.path else ''}{container.container_name}",
+            description=container.description,
+            env=self.env,
+        )
+
+    def _create_browse_paths(
+        self, entity: Union[DremioContainer, DremioDataset]
+    ) -> Optional[BrowsePathsV2Class]:
+        paths = []
+
+        if self.platform_instance:
+            paths.append(
+                BrowsePathEntryClass(
+                    id=self.platform_instance,
+                )
+            )
+
+        if entity.path:
+            for browse_path_level in range(len(entity.path)):
+                paths.append(
+                    BrowsePathEntryClass(
+                        id=entity.path[browse_path_level],
+                        urn=self.get_container_urn(
+                            name=entity.container_name
+                            if hasattr(entity, "container_name")
+                            else "",
+                            path=entity.path[: browse_path_level + 1],
+                        ),
+                    )
+                )
+
+        if paths:
+            return BrowsePathsV2Class(path=paths)
+        return None
+
+    def _create_container_class(
+        self, entity: Union[DremioContainer, DremioDataset]
+    ) -> Optional[ContainerClass]:
+        if entity.path:
+            return ContainerClass(
+                container=self.get_container_urn(
+                    path=entity.path,
+                    name="",

```

Comment:
why is name empty here ?

Reply:
This is because here dealing with a parent or intermediate container in the path hierarchy.

For eg.
If you have a path like ['level1', 'level2'], using name="" would result in a URN that identifies the container based purely on level1.level2

If you had used name="containerName", the URN might end up being level1.level2.containerName, which could be redundant if containerName isn't adding unique information